test/cql-pytest/cassandra_tests/validation/entities/uf_types_test.py

```diff
@@ -0,0 +1,345 @@
+        now = datetime.datetime.now()
+        # Cassandra only supports milliseconds in timestamps, so we need to round down to the nearest millisecond.
+        micros = now.microsecond // 1000 * 1000
+        type_defs = [
+            # udf type, table type, column, reference value
+            TypesTestDef("timestamp", "timestamp", "ts", now.replace(microsecond=micros)),
+            TypesTestDef("date", "date", "dt", Date(12345)),
+            TypesTestDef("time", "time", "tim", 12345),
+            TypesTestDef("uuid", "uuid", "uu", uuid.uuid4()),
+            TypesTestDef("timeuuid", "timeuuid", "tu", uuid.uuid1()),
+            TypesTestDef("tinyint", "tinyint", "ti", 42),
+            TypesTestDef("smallint", "smallint", "si", 43),
+            TypesTestDef("int", "int", "i", 44),
+            TypesTestDef("bigint", "bigint", "bi", 45),
+            TypesTestDef("float", "float", "f", 46.0),
+            TypesTestDef("double", "double", "d", 47.0),
+            TypesTestDef("boolean", "boolean", "x", True),
+            TypesTestDef("ascii", "ascii", "a", "tqbfjutld"),
+            TypesTestDef("text", "text", "txt", "k\u00f6lsche jung"),
+            # TypesTestDef(type, f"frozen<{type}>", "u", null),
+            TypesTestDef("tuple<int, text>", "frozen<tuple<int, text>>", "tup", (1, "foo")),
+        ]
+
+        schema = "(key int PRIMARY KEY"
+        values = [1]
+        for type_def in type_defs:
+            schema += ", " + type_def.column_name + ' ' + type_def.table_type
+            values.append(type_def.reference_value)
+        schema += ")"
+
+        with create_table(cql, test_keyspace, schema) as table:
+            insert_str = f"INSERT INTO {table} (key"
+            for type_def in type_defs:
+                insert_str += ", " + type_def.column_name
+            insert_str += ") VALUES (?"
+            for type_def in type_defs:
+                insert_str += ", ?"
+            insert_str += ")"
+            insert_stmt = cql.prepare(insert_str)
+            cql.execute(insert_stmt, values)
+
+            for i in range(len(values)):
+                values[i] = None
+            values[0] = 2
+            cql.execute(insert_stmt, values)
+
+            for type_def in type_defs:
+                use_wasm = is_scylla(cql, test_keyspace)
+                lang = "wasm" if use_wasm else "java"
+                fun_name = unique_name()
+                fun_wasm_src = read_function_from_file("test_types_with_and_without_nulls", f"check_arg_and_return_{type_def.column_name}", fun_name) if use_wasm else 'return input;'
+                fun_src = f"(input {type_def.udf_type}) CALLED ON NULL INPUT RETURNS {type_def.udf_type} LANGUAGE {lang} AS '{fun_wasm_src}'"
+                with new_function(cql, test_keyspace, fun_src, fun_name):
+                    # Cassandra returns timeuuids returned from UDFs as bytes (even though UUIDs are UUIDs), so if we're running Cassandra we should also compare the result to bytes.
```

Comment:
I don't understand, are you suggesting that this is a Cassandra bug (it returns the wrong type)?

Reply:
I think it is in fact a minor bug (even though the returned bytes correctly represent the UUID), but I didn't think it's enough to mark this case `cassandra_bug` because there it was simple enough to fix